src/main/java/edu/harvard/iq/dataverse/S3PackageImporter.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```

Comment:
Are we sure this code shouldn't go in StorageIO? This feels like the type of communication that we should be using it for.

Reply:
I kept it separate because the file mode code also keeps it separate. I thought that was also the better approach from the bits of discussion I had with others on the team. I would be up for moving it though if it seems right. I'm curious what @landreev @scolapasta think.

fyi this pr isn't representative of the actual changes, because it includes the dual mode code that has yet to be merged.